src/v/cloud_storage/partition_manifest.h

```diff
@@ -297,6 +297,14 @@ class partition_manifest : public base_manifest {
     partition_manifest truncate(model::offset starting_rp_offset);
     partition_manifest truncate();
 
+    /// \brief Truncate the manifest (remove entries from the manifest)
+    ///
+    /// \note this works the same way as 'truncate' but the 'archive' size
+    ///       gets correctly updated.
+    /// \param starting_rp_offset is a new starting offset of the manifest
+    /// \return manifest that contains only removed segments
+    partition_manifest spillover(model::offset starting_rp_offset);
```

Comment:
nit: might want to update the comments for `partition_manifest::truncate` above as it refers to spillover. Can be done in a follow up though.